tests/impact/client/test_client.py

```diff
@@ -156,3 +156,17 @@ def test_client_login_fail_lets_user_enter_new_key(_, login_fails):
     )
 
     cred_manager.get_key_from_prompt.assert_called()
+
+
+def test_client_login_api_key_empty_string_anon_login_and_dont_save_key(sem_ver_check):
```

Comment:
I would prefer if we could use BDD style acceptance criteria (given, when, then), e.g. "test_given_empty_login_key_when_login_anonymously_then_key_should_not_be_saved"

https://en.wikipedia.org/wiki/Behavior-driven_development